test/acceptance/tests/terminating-gateway/terminating_gateway_namespaces_test.go

```diff
@@ -0,0 +1,232 @@
+			}
+
+			releaseName := helpers.RandomName()
+			consulCluster := framework.NewHelmCluster(t, helmValues, ctx, cfg, releaseName)
+
+			consulCluster.Create(t)
+
+			consulClient := consulCluster.SetupConsulClient(t, c.secure)
+
+			t.Logf("creating Kubernetes namespace %s", testNamespace)
+			helpers.RunKubectl(t, ctx.KubectlOptions(), "create", "ns", testNamespace)
+			helpers.Cleanup(t, cfg.NoCleanupOnFailure, func() {
+				helpers.RunKubectl(t, ctx.KubectlOptions(), "delete", "ns", testNamespace)
+			})
+
+			staticClientNamespace := "ns2"
+			t.Logf("creating Kubernetes namespace %s", staticClientNamespace)
+			helpers.RunKubectl(t, ctx.KubectlOptions(), "create", "ns", staticClientNamespace)
+			helpers.Cleanup(t, cfg.NoCleanupOnFailure, func() {
+				helpers.RunKubectl(t, ctx.KubectlOptions(), "delete", "ns", staticClientNamespace)
+			})
+
+			ns1K8SOptions := &k8s.KubectlOptions{
+				ContextName: ctx.KubectlOptions().ContextName,
+				ConfigPath:  ctx.KubectlOptions().ConfigPath,
+				Namespace:   testNamespace,
+			}
+			ns2K8SOptions := &k8s.KubectlOptions{
+				ContextName: ctx.KubectlOptions().ContextName,
+				ConfigPath:  ctx.KubectlOptions().ConfigPath,
+				Namespace:   staticClientNamespace,
+			}
+
+			// Deploy a static-server that will play the role of an external service
+			t.Log("creating static-server deployment")
+			helpers.DeployKustomize(t, ns1K8SOptions, cfg.NoCleanupOnFailure, cfg.DebugDirectory, "../fixtures/bases/static-server")
+
+			// Register the external service
+			registerExternalService(t, consulClient, testNamespace)
+
+			// If ACLs are enabled we need to update the token of the terminating gateway
+			// with service:write permissions to the static-server service
+			// so that it can can request Connect certificates for it.
+			if c.secure {
+				updateTerminatingGatewayToken(t, consulClient, fmt.Sprintf(staticServerPolicyRulesNamespace, testNamespace))
+			}
+
+			// Create the config entry for the terminating gateway
+			createTerminatingGatewayConfigEntry(t, consulClient, ctx.KubectlOptions().Namespace, testNamespace)
+
+			// Deploy the static client
+			t.Log("deploying static client")
+			helpers.DeployKustomize(t, ns2K8SOptions, cfg.NoCleanupOnFailure, cfg.DebugDirectory, "../fixtures/cases/static-client-namespaces")
+
+			// If ACLs are enabled, test that intentions prevent connections.
+			if c.secure {
+				// With the terminating gateway up, we test that we can make a call to it
+				// via the static-server. It should fail to connect with the
+				// static-server pod because of intentions.
+				assertNoConnectionAndAddIntention(t, consulClient, ns2K8SOptions, testNamespace, testNamespace)
+			}
+
+			// Test that we can make a call to the terminating gateway
+			t.Log("trying calls to terminating gateway")
+			helpers.CheckStaticServerConnection(t, ns2K8SOptions, true, staticClientName, "http://localhost:1234")
+		})
+	}
+}
+
+const staticServerPolicyRulesNamespace = `namespace %s {
```

Comment:
```suggestion
const staticServerPolicyRulesNamespace = `namespace %q {
```

Surprised this doesn't need to be quoted.

Reply:
It did need to be quoted, the tests failed 😄 Good catch, thank you 🙏